src/auth/mod.ts

```diff
@@ -13,35 +15,169 @@ interface DecodedToken {
   };
 }
 
+interface JWK {
+  kid: string;
+  use: string;
+  kty: string;
+  alg: string;
+  n: string;
+  e: string;
+}
+
 const logger = getLogger(import.meta);
 
-function isSubInPodsConfig(payload: DecodedToken): boolean {
-  return podsConfig.pods.some((pod) =>
-    pod.sub === payload.sub &&
-    pod.serviceaccount.uid === payload["kubernetes.io"]?.serviceaccount.uid
-  );
+export async function verifyServiceAccountToken(
+  token: string,
+): Promise<string> {
+  const payload = await verifyToken(token);
+
+  if (
+    !payload["kubernetes.io"] || !payload["kubernetes.io"].serviceaccount.name
+  ) {
+    const message = "Payload does not contain service account name field";
+    logger.error(message);
+    throw new HTTPException(HTTP_STATUS_CODES.UNAUTHORIZED, {
+      message,
+    });
+  }
+
+  return payload["kubernetes.io"].serviceaccount.name;
+}
+
+let jwks: JWK[] | undefined;
+const jwkExpiration = 24 * 60 * 60 * 1000; // 24 hours
+let expirationTime = Date.now() + jwkExpiration;
+
+function isJWKExpired(): boolean {
+  return Date.now() > expirationTime;
```

Comment:
I see no advantage in having a split function for such simple check, except if it is very heavily used, otherwise inlining it simplifying the reading